code/TagField.php

```diff
@@ -173,4 +273,90 @@ public function saveInto(DataObjectInterface $record) {
 			$record->$name = implode(',', $values);
 		}
 	}
+
+	/**
+	 * Returns a JSON string of tags, for ajax-based search.
+	 *
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return string
+	 */
+	public function suggest(SS_HTTPRequest $request) {
+		if($this->recordClass === null) {
+			return '[]';
+		}
+
+		/**
+		 * @var DataObject $object
+		 */
+		$object = singleton($this->recordClass);
```

Comment:
I thought `singleton()` was limited to `DataObject`, but that's not the case of course. OK, then then inline declaration makes sense. Stylistically, I think block level comments should only be used for class and method level docs. In order to debug, I sometimes use my IDE to select a bunch of code and comment out via keyboard shortcuts and inline comments (`//`). Its not blocking my workflow, just irks me to see it - you'll be hard pressed finding many block level comments inside methods in core, so its one of those "implicit conventions".